backend/src/main/java/hu/flowacademy/konyhatunder/service/RecipeService.java

```diff
@@ -97,6 +102,73 @@ public Recipe createRecipe(String stringRecipe, MultipartFile image) throws Json
         return savedRecipeInRepository;
     }
 
+    public List<String> listRecipeDifficulty() {
+        List<String> difficulties = Arrays.stream(Difficulty.values()).map(Difficulty::getHungarianTranslation).collect(Collectors.toList());
+        log.debug("Listing all: {} difficulties", difficulties.size());
+        return difficulties;
+    }
+
+    public List<Recipe> listRecipesByIngredients(List<Ingredient> ingredientList) {
+        validateReceivedIngredients(ingredientList);
+        Set<Recipe> foundRecipes = new HashSet<>();
+        ingredientList.forEach(ingredient ->
+                foundRecipes.addAll(recipeRepository.findAllRecipesContainingIngredient(ingredient.getId())));
+        log.debug("Found {} recipe by criteria", foundRecipes.size());
+        return List.copyOf(foundRecipes);
+    }
+
+    public List<Recipe> listRecipesByCriteria(SearchByCriteriaDTO searchByCriteriaDTO) {
+        List<Recipe> foundRecipes = recipeRepository.findAll();
+        if (StringUtils.hasText(searchByCriteriaDTO.getName())) {
+            foundRecipes = foundRecipes.stream().filter(recipe -> recipe.getName().equalsIgnoreCase(searchByCriteriaDTO.getName())).collect(Collectors.toList());
+        }
+        if (searchByCriteriaDTO.getPreparationTime() != null) {
+            switch (searchByCriteriaDTO.getPreparationTime()) {
+                case 30:
```

Comment:
Mi van akkor ha a prepTime param 23? Ez itt vagy vvalójában nem egy prepTime a dto-ban és akkor rossz az elnevezése a változónak, és a típusa is, vagy valóban egy akármilyen integer bejöhet itt és akkor pedig az egész switch case felettéb problémás.

Reply:
Meg ahogy látom a kliens oldalról ezek az értékek jöhetnek be:
'30', '60', '120', '180', '240'

Reply:
FE küldi ezeket az értékeket, csak ezeket küldheti. Ha esetleg belenyúlna valaki, akkor a default rész futna. 
Ezek az értékek a prepTime felső határai egyébként. ki lehet listázni így a 0-29 perc köztieket, 30-59, 60-119, 120-179, vagy 180- felett. Viszotn itt is a korábbi szűrt lista van tovább szűrve. Query szinten tudjuk ezt kezelni?

Reply:
Itt ha nincs query szintre hozva a dolog akkor egy ENUM-ot használnék egy lower és upper bounddal plusz attributumnak.
Amúgy ha szabad a megadott érték akkor +-20-30 percet min 0-al.

Reply:
> FE küldi ezeket az értékeket, csak ezeket küldheti. Ha esetleg belenyúlna valaki, akkor a default rész futna.
> Ezek az értékek a prepTime felső határai egyébként. ki lehet listázni így a 0-29 perc köztieket, 30-59, 60-119, 120-179, vagy 180- felett. Viszotn itt is a korábbi szűrt lista van tovább szűrve. Query szinten tudjuk ezt kezelni?

Az intervallum paramot kell atadni a querynek azaz egy betweennel vagy sima relacioval szurni.

Reply:
Egyetértek: Ha ezek fix kategóriák, akkor szerintem elég lenne az Enum, de akkor a recepten is legyen az, nem szenvednék a tól-ig-gal, de most eleve szabad érték (?), ahol pedig kell a tól-ig, mert ez a logika valami UI logikának az értelmezése. Ha ott változna valami, akkor ide is át kellene vezetni, azaz most két helyre szedtétek szét. Az enumos változtatás és az intervallumos szűrés is egyértelmű helyzetet teremt.

Reply:
Módosítottuk. FE mostmár elküldi a két szélső értéket egy tömbben, és mi csak az alapján keresünk, ha nem volt korábbi szűrés, akkor a repository-ban, ha volt, akkor a korábbi listát tovább szűrve, az intervallum alapján.